test/module/irohad/ametsuchi/block_storage_test.cpp

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2018 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <boost/filesystem.hpp>
+#include <boost/filesystem/fstream.hpp>
+
+#include "ametsuchi/impl/block_storage_nudb.hpp"
+
+#include <gtest/gtest.h>
+#include "common/files.hpp"
+#include "common/types.hpp"
+
+#include "logger/logger.hpp"
+
+using namespace iroha::ametsuchi;
+using Identifier = BlockStorage::Identifier;
+
+static logger::Logger log_ = logger::testLog("BlockStore");
+
+class BlStore_Test : public ::testing::Test {
+ protected:
+  void clear() {
+    try {
+      namespace fs = boost::filesystem;
+      using perms = boost::filesystem::perms;
+      fs::permissions(
+          block_store_path,
+          perms::owner_read | perms::owner_write | perms::owner_exe);
+      fs::remove_all(block_store_path);
+    } catch (...) {
```

Comment:
![image](https://user-images.githubusercontent.com/750115/35767307-2075ef0a-08fb-11e8-94db-da3c618071f6.png)

Reply:
Normally you would want to remove test block store regardless of permissions or anything else. Here I do not bother myself which exceptions are thrown here, I don't care. 
This method `clean()` works exactly as I want: removes test dir or does nothing.